### ISSUE TITLE
Incorrect handling of parenthesized comparison operations

### ISSUE THREAD

**Issue by decorator-factory** [microsoft/pyright#890]
**Describe the bug**
In Python, `(a > b) > c` is just `(a > b) > c`, but `a > b > c` translates to `(a > b) and (b > c)`.
Pyright doesn't see the difference between `(a > b) > c` and `a > b > c`.

In this example, `A[T, U]` is a pair type. if `x: A[T, U]` and `y: A[U, V]`, then `z = x > y` is `A[T, V]` (constructed from the left element of `x` and right element of `y`). This is a toy example, but it demonstrates the point.

**To Reproduce**
1. Enter the code into the editor
2. Hover over `a` to see that its inferred type is `A[str, complex]`
3. Hover over `b` to see that is has the same type
4. Run the code and see that `b` is `A('y', (1+0j))` (inferred correctly)
5. Run the code and see that `a` is `A(1, (1+0j))` (inferred incorrectly)

**Code**
```py
from typing import Generic, TypeVar

T = TypeVar("T")
U = TypeVar("U")
V = TypeVar("V")

class A(Generic[T, U]):
    left: T
    right: U
    def __init__(self, left: T, right: U):
        self.left = left
        self.right = right

    def __repr__(self):
        return f"A({self.left!r}, {self.right!r})"

    def __ge__(self: "A[T, U]", other: "A[U, V]") -> "A[T, V]":
        return A(self.left, other.right)

x: A[int, str] = A(1,  "a")
y: A[str, float] = A("y", 3.14)
z: A[float, complex] = A(2.718, 1+0j)

a = (x >= y) >= z
b = (x >= y) and (y >= z)

print(a)
print(b)
```

**VS Code extension or command-line**
VS Code extension

**P.S.**
If you do:
```py
xy = x >= y
a = xy >= z
```
Pyright infers the type correctly

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of Pyright & Pylance.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.58, which I just published. It will also be included in the next version of Pylance.
